### PR TITLE
Fix global step increment on training_epoch_end

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

global_step gets incremented if training_epoch_end is implemented. 
This shouldn't be necessary and will lead to a misalignment in logs. 

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by rohitgr7** [comment]
Looks like all tests are passing. Maybe add a test too to make sure its all good in future 🙂

**Comment by awaelchli** [comment]
That's the plan :)

**Comment by awaelchli** [comment]
William will solve this in his branch

**Comment by williamFalcon** [comment]
The horovod test is failing randomly.... also verified on my local.

image


@tgaddair  mind taking a look in a follow on PR?
merging this one to unblock releases.

File: tests/models/test_horovod.py [hunk 1]
```diff
@@ -195,35 +195,34 @@ def get_optimizer_params(optimizer):
...
-
-    adjusted_lr1 = [pg['lr'] for pg in trainer.optimizers[0].param_groups][0]
-    adjusted_lr2 = [pg['lr'] for pg in trainer.optimizers[1].param_groups][0]
-
-    # Called ones after end of epoch with gamma=0.1
-    assert pytest.approx(init_lr * 0.1) == adjusted_lr1
-
-    # Called every 3 steps, meaning for 1 epoch of 11 batches, it is called 3 times with gamma=0.1
-    assert pytest.approx(init_lr * 0.1) == adjusted_lr2
+# @pytest.mark.skipif(platform.system() == "Windows", reason="Horovod is not supported on Windows")
```
Comment (williamFalcon) [inline tests/models/test_horovod.py:198]: @tgaddair weird error... happens only on some machines some times haha

**Pull request merged**